File: 2015/zig/day1.zig
const std = @import("std");
const data = @embedFile("day1.txt");

pub fn part1() !void {
    //    var ch = std.mem.split(u8, data, "");
    //   while (ch.next()) |x| {
    //      std.debug.print("{s}", .{x});
    // }
    var floor: i16 = 0;
    for (data) |x| {
        if (x == '(') {
            floor += 1;
        } else if (x == ')') {
            floor -= 1;
        }
    }
    std.debug.print("{d}\n", .{floor});
}

pub fn part2() !void {
    //    var ch = std.mem.split(u8, data, "");
    //   while (ch.next()) |x| {
    //      std.debug.print("{s}", .{x});
    // }
    var floor: i16 = 0;
    var position: i16 = 0;
    for (data) |x| {
        position += 1;
        if (x == '(') {
            floor += 1;
        } else if (x == ')') {
            floor -= 1;
        }
        if (floor == -1) {
            break;
        }
    }
    std.debug.print("{d}\n", .{position});
}

pub fn main() !void {
    try part1();
    try part2();
}

File: 2015/go/day1.go
package main

import (
	"fmt"
	"io/ioutil"
)


func main()  {
    part1()
    part2()
}

func part1() {
    data, err := ioutil.ReadFile("inputs/day1.txt")
    if err != nil {
        fmt.Printf("%v", err)
    }
    floor := 0
    for _, ch := range string(data) {
        if ch == '(' {
            floor += 1
        }
        if ch == ')' {
            floor -= 1
        }
        
    }
    fmt.Printf("%d\n", floor)

}
func part2() {
    data, err := ioutil.ReadFile("inputs/day1.txt")
    if err != nil {
        fmt.Printf("%v", err)
    }
    floor := 0
    position := 0
    for _, ch := range string(data) {
        position += 1
        if ch == '(' {
            floor += 1
        }
        if ch == ')' {
            floor -= 1
        }

        if floor == -1 {
            fmt.Printf("%d\n", position)
            break
        }

    }
}
